crates/nu-command/src/filesystem/save.rs

```diff
@@ -121,40 +119,102 @@ impl Command for Save {
                     force,
                 )?;
 
-                match (stdout, stderr) {
-                    (Some(stdout), stderr) => {
-                        // delegate a thread to redirect stderr to result.
-                        let handler = stderr
-                            .map(|stderr| match stderr_file {
-                                Some(stderr_file) => thread::Builder::new()
-                                    .name("stderr redirector".to_string())
-                                    .spawn(move || {
-                                        stream_to_file(stderr, stderr_file, span, progress)
-                                    }),
-                                None => thread::Builder::new()
-                                    .name("stderr redirector".to_string())
-                                    .spawn(move || stderr.drain()),
-                            })
-                            .transpose()
-                            .err_span(span)?;
-
-                        let res = stream_to_file(stdout, file, span, progress);
-                        if let Some(h) = handler {
-                            h.join().map_err(|err| ShellError::ExternalCommand {
-                                label: "Fail to receive external commands stderr message"
-                                    .to_string(),
-                                help: format!("{err:?}"),
-                                span,
-                            })??;
+                // https://github.com/nushell/nushell/pull/9377 contains the reason
+                // for not using BufWriter<File>
+                //
+                // TODO: flag to specify buffered (BufWriter), line buffered (LineWriter), or no buffering (File).
```

Comment:
I'm not sure this will ever be necessary, because the input stream is generally buffered in some way, even if by the command that produced it. If someone knows they want to use bigger buffers, it might be better to provide a command to adapt the stream with `read_exact()` or `read_until()` calls to ensure the desired chunking behaviour. And of course it should already be possible to do `| lines | save` if you definitely want line buffering.

I like this behaviour though, it means that the output of an external command gets written to file with the flushing behaviour it was intended to have, which is particularly good if saving to something like a FIFO where the interactivity actually matters.

Reply:
Oh yeah, that's a good observation! There might be an argument to add the flags for performance reasons (e.g., to avoid the allocations), but it's probably not worth it / it won't make a noticeable difference.